RELEASE.md

```diff
@@ -11,7 +11,8 @@ The release process consists of these steps:
      the staging repository closed, and the artifacts on the way to Maven Central (it takes 20min+ to get there).
      In case of failures it is safe to retry by restarting the CROSSDOCK step of the build. If it keeps
      failing, sometimes it may be necessary to close the staging repository manually.
-  1. Once the artifacsts are available on Maven
+  1. Create a release on GitHub for the new tag. Use the changes from the `CHANGELOG` as the description.
```

Comment:
I think this is redundant, we should only do Github release and not do steps 3-4 above

Reply:
Agreed. For this commit, I wanted to quickly fix the version issue to get the CI working again and wanted to document that we are indeed using GH's release feature.

During next release, I'll record the exact steps and change the instructions.